### ISSUE TITLE
"invalid number" warning should be different when only scientific notation is allowed

### ISSUE THREAD

**Issue by christianp** [numbas/Numbas#650]
At the moment, if you enter an invalid number in a number entry part, you're shown the warning message "Your answer is invalid. You must enter an integer or a decimal."

This isn't quite correct when only scientific notation is allowed. The message should be one of:

* "You must enter a number in scientific notation" (not helpful if you think you already have)
* "You must enter a significand followed by the letter 'e' and an exponent" (uses a couple of words students might not recognise; some lecturers might be pedantic about upper or lower case 'e')
* "Your answer is invalid." (Doesn't give any hint at all)

I think the first one is the safest.

**Comment by orsim** [comment]
I agree the first option is the best - but will the full message retain the first sentence "Your answer is invalid. [...]"? If so, this may be a bit too "harsh", after all at that stage the student may not have finished and the answer may be "partially" valid.

How about dropping that first sentence and go with "Warning: you must enter a number in scientific notation" or something similar?

**Comment by ejpetrie** [comment]
perhaps include an example of a number in scientific notation?
